src/unstract/sdk/file_storage/fs_permanent.py

```diff
@@ -0,0 +1,42 @@
+from typing import Union
+
+from fsspec import AbstractFileSystem
+
+from unstract.sdk.exceptions import FileStorageError
+from unstract.sdk.file_storage.constants import Common
+from unstract.sdk.file_storage.fs_impl import FileStorage
+from unstract.sdk.file_storage.fs_provider import FileStorageProvider
+
+
+class PermanentFileStorage(FileStorage):
+    SUPPORTED_FILE_STORAGE_TYPES = [
+        FileStorageProvider.GCS.value,
+        FileStorageProvider.S3.value,
+        FileStorageProvider.Azure.value,
+    ]
+
+    def __init__(self, provider, credentials: dict() = None):
```

Comment:
@gaya3-zipstack is adding type hint with brackets correct? `dict()` . I haven't observed this before hence asking

Reply:
Please run the mypy checks once @gaya3-zipstack - I think you might have to change the type annotation as `dict[str, Any]`. Also please annotate the `provider` and the return type as well

Reply:
Addressed. Thank you. I forgot that mypy is disabled.